bliss/simulator/prior.py

```diff
@@ -95,18 +99,72 @@ def __init__(
         self.galaxy_a_scale = galaxy_a_scale
         self.galaxy_a_bd_ratio = galaxy_a_bd_ratio
 
-    def sample_prior(self) -> TileCatalog:
+        # load all star, galaxy ratios required for sampling
+        if self.n_bands == 1:
+            return
+        for field_params in sdss_fields["field_list"]:
+            run = field_params["run"]
+            camcol = field_params["camcol"]
+            fields = field_params["fields"]
+            for field in fields:
+                sdss_path = Path(self.sdss)
+                camcol_dir = sdss_path / str(run) / str(camcol) / str(field)
+                po_path = camcol_dir / f"photoObj-{run:06d}-{camcol:d}-{field:04d}.fits"
+                po_fits = fits.getdata(po_path)
+
+                fluxes = column_to_tensor(po_fits, "psfflux")
+                objc_type = column_to_tensor(po_fits, "objc_type").numpy()
+                thing_id = column_to_tensor(po_fits, "thing_id").numpy()
+
+                star_fluxes_rest = []
+                gal_fluxes_rest = []
+
+                for obj in objc_type:
+                    if obj == 6 and thing_id[obj] != -1 and torch.all(fluxes[obj] > 0):
+                        ratios = (  # noqa: WPS220
+                            fluxes[obj][self.bands[1] :] / fluxes[obj][self.bands[0]]
+                        )
+                        star_fluxes_rest.append(ratios)  # noqa: WPS220
+                    elif obj == 3 and thing_id[obj] != -1 and torch.all(fluxes[obj] > 0):
+                        ratios = (  # noqa: WPS220
+                            fluxes[obj][self.bands[1] :] / fluxes[obj][self.bands[0]]
+                        )
+                        gal_fluxes_rest.append(ratios)  # noqa: WPS220
+
+                if star_fluxes_rest:
+                    self.rcf_stars_rest[(run, camcol, field)] = star_fluxes_rest
+                if gal_fluxes_rest:
+                    self.rcf_gals_rest[(run, camcol, field)] = gal_fluxes_rest
+
+    def sample_prior(self, batch_rcfs) -> TileCatalog:
         """Samples latent variables from the prior of an astronomical image.
 
+        Args:
+            batch_rcfs: run, camcol, field combinations needed for sampling.
+
         Returns:
             A dictionary of tensors. Each tensor is a particular per-tile quantity; i.e.
             the first three dimensions of each tensor are
             `(batch_size, self.n_tiles_h, self.n_tiles_w)`.
             The remaining dimensions are variable-specific.
         """
         locs = self._sample_locs()
-        galaxy_params = self._sample_galaxy_prior()
-        star_fluxes = self._sample_star_fluxes()
+        select_gal_rcfs = []
+        select_star_rcfs = []
+
+        for rcf in batch_rcfs:
```

Comment:
(L155 - 167) The relative flux ratios corresponding to the selected RCFs for this particular batch are collected and apssed to `self._sample_galaxy_prior` and `self._sample_star_fluxes`. Each image that is generated samples flux ratios from one particular FITs file.